packages/noya-state/src/reducers/application.ts

```diff
@@ -1257,6 +1277,231 @@ export function reducer(
+            if (!attributes.MSAttributedStringColorAttribute) {
+              attributes.MSAttributedStringColorAttribute = color;
+              return;
+            }
+
+            attributes.MSAttributedStringColorAttribute.alpha = color.alpha;
+            attributes.MSAttributedStringColorAttribute.blue = color.blue;
+            attributes.MSAttributedStringColorAttribute.red = color.red;
+            attributes.MSAttributedStringColorAttribute.green = color.green;
+          }
+        });
+      });
+    }
+    case 'setTextFontName': {
+      const [, value] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded =
+            layer.style?.textStyle?.encodedAttributes
+              .MSAttributedStringFontAttribute.attributes;
+          const attributes =
+            layer.attributedString.attributes[0].attributes
+              .MSAttributedStringFontAttribute.attributes;
+
+          if (!encoded) return;
+
+          const face = encoded.name.split('-')[1]
+            ? '-' + encoded.name.split('-')[1]
+            : '';
+
+          const newName = value.startsWith('-')
+            ? encoded.name.split('-')[0] + value
+            : value + face;
+
+          encoded.name = newName;
+          if (attributes) attributes.name = newName;
+        });
+      });
+    }
+    case 'setTextFontSize': {
+      const [, value] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded =
+            layer.style?.textStyle?.encodedAttributes
+              .MSAttributedStringFontAttribute.attributes;
+          const attributes =
+            layer.attributedString.attributes[0].attributes
+              .MSAttributedStringFontAttribute.attributes;
+
+          if (encoded) encoded.size = value;
+          if (attributes) attributes.size = value;
+        });
+      });
+    }
+    case 'setTextLetterSpacing':
+    case 'setTextLineSpacing':
+    case 'setTextParagraphSpacing': {
+      const [, value] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded = layer.style?.textStyle?.encodedAttributes;
+          const attributes = layer.attributedString.attributes[0].attributes;
```

Comment:
I wonder if we could do something like (and this might not work, due to type differences):

```ts
const allAttributes = [encoded, ...layer.attributedString.attributes]

allAttributes.forEach((attribute) => {
  switch (action[0]) {
    case 'setTextLetterSpacing': {
      attribute.kerning = value;
      break;
    }
  }
})
```

Reply:
I did something like it but I'm not sure if it's optimal I added the "I am not sure if it's optimal comment"